indra/newview/llselectmgr.cpp

```diff
@@ -2338,21 +2350,20 @@ void LLSelectMgr::selectionSetAlphaGamma(U8 gamma)
         f(const U8 &t) : mAlphaGamma(t) {}
         bool apply(LLViewerObject *object, S32 te)
         {
-            if (object->permModify())
+            bool can_modify = object->permModify();
+            if (can_modify || object->permYouOwner())
             {
                 // update viewer side color in anticipation of update from simulator
                 object->setTEAlphaGamma(te, mAlphaGamma);
             }
-            else
-            {
-                packAlphaGammaOverride(object);
-            }
             return true;
         }
     } setfunc(gamma);
     getSelection()->applyToTEs(&setfunc);
 
-    LLSelectMgrSendFunctor sendfunc;
+    // TODO: remove LLSelectMgrAlphaGammaBypassFunctor and instead
+    // use the normal LLSelectMgrSendFunctor and enhance the server to know when to allow bypass
+    LLSelectMgrAlphaGammaBypassFunctor sendfunc;
     getSelection()->applyToObjects(&sendfunc);
 }
```

Comment:
That was the wrong place to call `packAlphaGammaOverride()`.  By putting it there the viewer would send one message for every object face!

The correct way to do it is to first make ALL of the changes to the local object (if permission rules allow) and only when all changes made: send an update to the server.  The server validates the changes and will always send a confirmation or denied update back to the viewer that tried to make the change.

Reply:
Noted!  Thank you!